src/plugins/intel_cpu/src/graph.cpp

```diff
@@ -193,122 +191,48 @@ void Graph::Replicate(const std::shared_ptr<const ov::Model> &subgraph) {
-        for (size_t parentPort = 0; parentPort < parentOp->get_output_size(); parentPort++) {
-            if (childOp->input(childInputPort).get_tensor_ptr() == parentOp->output(parentPort).get_tensor_ptr()) {
-                return static_cast<int>(parentPort);
-            }
-        }
-
-        return -1;
+    auto hasSubgraphConsumers = [](const NodePtr& node) -> bool {
+        const auto& childEdges = node->getChildEdges();
+        return std::any_of(childEdges.begin(), childEdges.end(), [](const EdgeWeakPtr& edge) -> bool {
+            auto edgePtr = edge.lock();
+            if (!edgePtr)
+                return false;
+            return edgePtr->getChild()->getType() == Type::Subgraph;
+        });
     };
 
-    OV_ITT_SCOPE_NEXT(FIRST_INFERENCE, taskChain, "AllNodes");
-
-    // Replicate All Nodes in topological order
-    for (const auto& op : orderedOps) {
-        const NodePtr node(Node::factory().create(op, context));
-
-        graphNodes.push_back(node);
-
-        if (op->get_type_info() == op::v0::Parameter::get_type_info_static()) {
-            const auto inInfo = inputsInfo.find(node->getName());
-            if (inInfo != inputsInfo.end()) {
-                inputNodesMap[node->getName()] = node;
-                if (node->isDynamicNode()) {
-                    graphHasDynamicInput = true;
-                }
-            }
-        }
-
-        if (op->get_type_info() == op::v0::Result::get_type_info_static()) {
-            const auto &input = op->input_value(0);
-            const auto name = op::util::get_ie_output_name(input);
-
-            if (outputsInfo.count(name) != 0) {
-                outputNodesMap[name] = node;
-            }
-        }
-
-        op2node[op] = node;
-
-        for (size_t port = 0; port < op->get_input_size(); port++) {
-            auto parentOp = op->get_input_node_shared_ptr(port);
-            auto parentNode = op2node[parentOp];
-
-            EdgePtr edge(new Edge(parentNode, node, getParentOutputPort(op, parentOp, port), static_cast<int>(port)));
-            node->addEdge(edge);
-            graphEdges.push_back(edge);
-        }
-
-        if (!one_of(op->get_type_info(),
-                op::v0::Result::get_type_info_static(),
-                op::v3::Assign::get_type_info_static(),
-                op::v6::Assign::get_type_info_static())) {
-            for (size_t oi = 0; oi < op->get_output_size(); oi++) {
-                if (op->get_output_target_inputs(oi).empty()) {
-                    unusedOutputs.push_back(op->output(oi));
-                }
-            }
+    auto find_input_port_prec = [&](const std::string& name) -> ov::element::Type_t {
+        for (auto& it : subgraph->inputs()) {
+            auto port_name = get_port_name(it, is_legacy_api);
+            if (port_name == name)
+                return it.get_element_type();
+            port_name = it.get_node_shared_ptr()->get_friendly_name();
```

Comment:
get_port_name() call ov::op::util::get_ie_output_name() to get port name, the latter will first to call get_ov_tensor_legacy_name(), if cannot get it will continue call get_friendly_name().
 Sometimes (seldom test cases), get_port_name() will get different names because legacy name it set, so we need search friendly name if cannot match the name.
BTW, there are several names, such as legacy_name, friendly_name, tensor_name, unique_name, etc, we should refactor to get some unified names.